gwpy/plot/colorbar.py

```diff
@@ -155,67 +148,112 @@ def _scale_height(value, ax):
-        'borderpad': 0.,
-    }
-
-    # get orientation based on location
-    if location.lower() in ('left', 'right'):
-        pad = kwargs.pop('pad', _scale_width(.1, ax))
-        kwargs.setdefault('orientation', 'vertical')
-    elif location.lower() in ('top', 'bottom'):
-        pad = kwargs.pop('pad', _scale_height(.1, ax))
-        kwargs.setdefault('orientation', 'horizontal')
-    orientation = kwargs.get('orientation')
-
-    # set params for orientation
-    if orientation == 'vertical':
-        inset_kw['width'] = .12
-        inset_kw['height'] = '100%'
+    # calculate default width and padding for the relevant orientation
+    orientation = "vertical" if location in ('left', 'right') else "horizontal"
+    if orientation == "vertical":
+        size = _scale_width(.1, ax)
+    elif orientation == "horizontal":
+        size = _scale_height(.1, ax)
+    pad = size if pad is None else pad
+    width = size if width is None else width
+
+    # where to start on the long axis (for centre-aligned Axes)
+    l0 = .5 - length / 2.
+
+    if location == "left":
+        return 0 - pad - width, l0, width, length
+    if location == "right":
+        return 1 + pad, l0, width, length
+    if location == "top":
+        return l0, 1 + pad, length, width
+    # "bottom":
+    return l0, 0 - pad - width, length, width
+
+
+def _make_inset_axes(
+    ax,
+    location='right',
+    width=0.012,
+    length=1.,
+    pad=None,
+    **kwargs,
+):
+    """Create a new `Axes` to support a colorbar using `Axes.inset_axes`.
+    """
+    # set default orientation
+    if location in ('left', 'right'):
+        orientation = kwargs.setdefault("orientation", "vertical")
+    elif location in ('top', 'bottom'):
+        orientation = kwargs.setdefault("orientation", "horizontal")
     else:
-        inset_kw['width'] = '100%'
-        inset_kw['height'] = .12
-
-    # set location and anchor position based on location name
-    # NOTE: matplotlib-1.2 requres a location code, and fails on a string
-    #       we can move back to just using strings when we drop mpl-1.2
-    inset_kw['loc'], inset_kw['bbox_to_anchor'] = {
-        'left': (LOC_CODES['lower right'], (-pad, 0., 1., 1.)),
-        'right': (LOC_CODES['lower left'], (1+pad, 0., 1., 1.)),
-        'bottom': (LOC_CODES['upper left'], (0., -pad, 1., 1.)),
-        'top': (LOC_CODES['lower left'], (0., 1+pad, 1., 1.)),
-    }[location]
-
-    # allow user overrides
-    for key in filter(inset_kw.__contains__, kwargs):
-        inset_kw[key] = kwargs.pop(key)
-
-    return inset_axes(ax, **inset_kw), kwargs
+        raise ValueError(f"inset Axes location '{location}' not recognised")
+
+    # set default ticklocation for the right location/orientation
+    if location == "top" and orientation == "horizontal":
+        kwargs.setdefault("ticklocation", "top")
+    if location == "left" and orientation == "vertical":
+        kwargs.setdefault("ticklocation", "left")
```

Comment:
what happens if location == "bottom" or "right"?

Reply:
Then the 'default' `ticklocation` is used, which is `bottom` or `right` respectively.